src/index.ts

```diff
@@ -1,21 +1,28 @@
-import * as core from '@actions/core';
-import * as exec from '@actions/exec';
-import path from 'path';
+import path from "path";
+import url from "url";
+import * as core from "@actions/core";
+import * as exec from "@actions/exec";
 
-async function run(): Promise<void> {
-  try {
-    const parentDir = path.resolve(__dirname, '..');
-    
-    // Get action inputs
-    process.env.INPUT_UNSET_PREVIOUS = core.getInput('unset-previous');
-    process.env.INPUT_EXPORT_ENV = core.getInput('export-env');
+const run = async () => {
+	try {
+		const currentFile = url.fileURLToPath(import.meta.url);
+		const currentDir = path.dirname(currentFile);
+		const parentDir = path.resolve(currentDir, "..");
 
-    // Execute bash script
-    await exec.exec(`sh -c "` + parentDir + `/entrypoint.sh"`);
+		// Get action inputs
+		process.env.INPUT_UNSET_PREVIOUS = core.getInput("unset-previous");
+		process.env.INPUT_EXPORT_ENV = core.getInput("export-env");
 
-  } catch (error: any) {
-    core.setFailed(error.message);
-  }
-}
+		// Execute bash script
+		await exec.exec(`sh -c "` + parentDir + `/entrypoint.sh"`);
+	} catch (error) {
+		// https://kentcdodds.com/blog/get-a-catch-block-error-message-with-typescript
```

Comment:
Maybe a silly nit, but adding some context for this link may help a bit.
The link itself may be explanatory, but since it's not from some official documentation (e.g. https://www.typescriptlang.org/docs/) I may want to know why we use this source as a reference.

Reply:
@edif2008 - Not a silly nit at all! For context, Kent C. Dodds in a well-known JavaScript developer and a trustworthy reference 🥇

I've added a short comment to add some context on what this code does in addition to the link to KCD's article. I also re-checked his article and noticed that I missed his edge case example where he addresses if something modifies the `Error` constructor, and so I've added that missing piece here too 👍

Reply:
Resolved via https://github.com/1Password/load-secrets-action/pull/34/commits/0eebc66811c3675d2436786f531f3f00d8969883 commit.